src/ert/_c_wrappers/enkf/site_config.py

```diff
@@ -1,74 +1,119 @@
 import os
+from dataclasses import dataclass
 
-from cwrap import BaseCClass
-
-from ert._c_wrappers import ResPrototype
+from ert._c_wrappers.config import ConfigContent
 from ert._c_wrappers.enkf.config_keys import ConfigKeys
-from ert._c_wrappers.job_queue import EnvironmentVarlist, ExtJob
-
-
-class SiteConfig(BaseCClass):
-    TYPE_NAME = "site_config"
-    _alloc = ResPrototype("void* site_config_alloc(config_content)", bind=False)
-    _site_config_alloc_default = ResPrototype(
-        "void* site_config_alloc_default()", bind=False
-    )
-    _alloc_full = ResPrototype(
-        "void* site_config_alloc_full(ext_joblist, env_varlist)", bind=False
-    )
-    _free = ResPrototype("void site_config_free( site_config )")
-    _get_installed_jobs = ResPrototype(
-        "ext_joblist_ref site_config_get_installed_jobs(site_config)"
-    )
-    _get_license_root_path = ResPrototype(
-        "char* site_config_get_license_root_path(site_config)"
-    )
-    _get_location = ResPrototype("char* site_config_get_location()", bind=False)
-    _get_config_file = ResPrototype("char* site_config_get_config_file(site_config)")
-    _get_env_var_list = ResPrototype(
-        "env_varlist_ref site_config_get_env_varlist(site_config)"
-    )
-
-    def __init__(self, config_content=None):
-
-        if config_content:
-            c_ptr = self._alloc(config_content)
-        else:
-            c_ptr = self._site_config_alloc_default()
-
-        if c_ptr is None:
-            raise ValueError("Failed to construct SiteConfig instance.")
-
-        super().__init__(c_ptr)
+from ert._c_wrappers.job_queue import EnvironmentVarlist, ExtJob, ExtJoblist
+
+
+@dataclass
+class SiteConfig:
+    job_list: ExtJoblist
+    env_vars: EnvironmentVarlist
```

Comment:
Why not?
```
@dataclass
class SiteConfig:
    job_list: ExtJoblist = ExtJoblist()
    env_vars: EnvironmentVarlist = EnvironmentVarlist()
```

Reply:
What will that do? Does it make a global object?

Reply:
Setting the default value will have to be done with
```python
from dataclasses import dataclass, field

@dataclass
class SiteConfig:
    job_list: ExtJoblist = field(default_factory=ExtJoblist)
    env_vars: EnvironmentVarlist = field(default_factory=EnvironmentVarlist)
```
to avoid every defaulted `job_list` being set to the same `ExtJobList`.